### PR TITLE
supporting file type from sonatype nexus query urls

### PR DESCRIPTION
**Pull request by guyyosan**
sonatype nexus sometimes uses urls with query params.

attempt to fix: https://github.com/CocoaPods/cocoapods-downloader/issues/100

tested locally with sonatype nexus, it works

### CODE REVIEW

**Comment by dnkoutso** [comment]
Can you please add a test?

**Comment by guyyosan** [comment]
@dnkoutso I'm a novice ruby dev :)
can you please refer me to this method's test: type_with_url?

**Comment by guyyosan** [comment]
@dnkoutso fixtures? ex. https://github.com/guyyosan/cocoapods-downloader/blob/master/spec/remote_file_spec.rb ?

**Comment by dnkoutso** [comment]
Yes add one in `remote_file_spec.rb`.

Run it as `bundle exec bacon /path/to/remote_file_spec.rb`

**Comment by guyyosan** [comment]
@dnkoutso can you please help me understand the ci fails?
not sure it's coming from my test

**Comment by guyyosan** [comment]
@dnkoutso it's working ok now. build is passing on 2/4 machines. the other 2 machines has issues with gem installations not related to my fix.

**Comment by seanreinhardtapps** [comment]
The failed tests are an issue with RVM with certain Ruby versions on Travis CI VMs. See #97

**Comment by guyyosan** [comment]
@seanreinhardtapps @dnkoutso is there a way to rerun ci now that https://github.com/CocoaPods/cocoapods-downloader/pull/97 is merged?
I mean, without adding a new commit

File: lib/cocoapods-downloader/remote_file.rb [hunk 1]
```diff
@@ -58,6 +58,28 @@ def should_flatten?
 
       def type_with_url(url)
         case URI.parse(url).path
+        when /\.zip$/
```
Comment (amorde) [inline lib/cocoapods-downloader/remote_file.rb:61]: could this get pulled out into a separate function so the `case` logic isn't duplicated in `type_with_url_query`? maybe `type_with_extension` and then call that here and in `type_with_url_query`

**Comment by amorde** [comment]
LGTM with 1 minor change

**Comment by guyyosan** [comment]
@amorde updated

**Comment by guyyosan** [comment]
@amorde Hi Eric,
last build is failing for one machine.. doesn't look code related:

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Comment by guyyosan** [comment]
@dnkoutso can you please check this, I really need a proxy working asap and this is a blocker.
The build is actually succeeds on all machines but 1 and I don't understand the error, it's like output is missing

**Pull request closed without merging**